File: src/test/fixtures/kotlin/magicNumber.kt
const val MAX_RETRIES = 5

fun cleanCommonValues(x: Int): Int {
    val total = x * 1
    return total + 0
}

fun flaggedMagicNumbers(price: Double): Double {
    var total = price * 1.08
    if (total > 50) {
        total += 15.75
    }
    return total
}

fun exemptIndexAndDefault(arr: IntArray, weight: Int = 42): Int {
    val first = arr[0]
    return first + weight
}

fun cleanNegativeValue(flag: Boolean): Int {
    if (flag) {
        return -1
    }
    return 1
}

File: src/test/fixtures/csharp/MagicNumber.cs
class CleanCommonNumbers
{
    const int MaxRetries = 5;
// clean — not flagged by magic number
    static int CleanCommonValues(int x) => x * 1;
}

class CleanNegativeNumbers
{
// clean — not flagged by magic number
    static int CleanNegativeValue(bool flag) => flag ? -1 : 1;
}

class FlaggedNumbers
{
// flagged — magic number
    static double FlaggedMagicNumbers(double price)
    {
        double total = price * 1.08;
        return total > 50 ? total + 15.75 : total;
    }
}
